### DISCUSSION TITLE
Mod Management for a Server Utility Tool

### DISCUSSION THREAD

**Discussion by alteredhero** [ebkr/r2modmanPlus#1084] [Q&A] [answered]
I'm currently working on a project for me and my friends where my friends will be able to restart and update both the mods and the server that runs on my server pc so I do not have to be present in order to do so. The piece of this puzzle I'm working on currently, and seems to be the hardest portion, is the ability to update the mods remotely. At this time, I'm able to download and move the mods. But it seems that my understanding of which folders need to go where is amiss. And since not everyone follows the same structure when they upload their mods to thunderstore it seems, I've run into quite the predicament.

So what I'm trying to find out, is if its possible to interact with r2modman remotely using scripts to do what I need. Or, if its possible to see how r2modman figures out where files need to go.

I can provide my current script for analysis as well.

**Comment by MythicManiac** [comment, marked as answer]
I'm afraid there's no good support for this use _yet_, though it is something we're working towards long term. https://github.com/thunderstore-io/thunderstore-cli repo is intended to replace the main "business logic" of mod management eventually, but it's not quite there yet (and importantly right now it still diverges from what r2modman does, so you can't use it out of the box.)

**Comment by alteredhero** [comment]
Okay, I'll keep that in mind then. I think i'll be able to get what i want working with what I have, its just going to take longer than i anticipated. But I guess thats just how developing something goes.
